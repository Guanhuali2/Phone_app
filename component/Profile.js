import React, { Component } from 'react';
import { StyleSheet, Image,Text, View } from 'react-native';
import {ProfileFetch} from './ProfileFetch.js'
/*
View page for Profile
*/
class Profile extends Component {
  constructor(props) {
    super(props);
    const accessToken = '';
    this.profile = new ProfileFetch(accessToken)

    this.state = {
      loading: true
    }
    this.getFile();
  }

  async getFile(){
    const temp = await this.profile.file();
    if (temp == "failed"){
      this.setState({ hasError: true });
      return;
    }
    const body = this.profile.getState();
    this.setState({
      loading: false
    })
    try{
      this.setState({
        name: body.name,
        bio: body.bio,
        twitter: body.twitter,
        avatarUrl: body.avatarUrl,
        location: body.location,
        url: body.url,
        create: body.created,
        countres: body.countres,
        countfw: body.countfw,
        countfi: body.countfi
      })
    }
    catch(error){
      this.setState({ hasError: true });
    }
  }
  /*
  render the view page
  */
  render() {
    let normalView;
    //check if the page has error or not
    if(this.state.hasError){
      //if error found, it will show invalid warning
      normalView = (<View>
        <Text key='names' style={styles.titleText}>Invalid Profile Page</Text>
      </View>)
    }else{
      normalView = (<View style={styles.container}>
        <Image style={styles.tinyLogo} source={this.state.avatarUrl} />
        <Text key='names' style={styles.titleText}>{this.state.name}</Text>
        <Text key='bio' style={styles.titleText}>{this.state.bio}</Text>
        <Text key='username' style={styles.titleText}>{this.state.twitter}</Text>
        <Text key='location' style={styles.titleText}>{this.state.location}</Text>
        <Text key='website' style={styles.titleText}>{this.state.url}</Text>
        <Text key='create' style={styles.titleText}>{this.state.create}</Text>
        <Text key='countcou' style={styles.titleText}>{this.state.countres}</Text>
        <Text key='countfw' style={styles.titleText}>{this.state.countfw}</Text>
        <Text key='countfi' style={styles.titleText}>{this.state.countfi}</Text>
      </View>)
    }
    return (
      <View style={styles.container}>
        {this.state.loading ? <Text style={styles.titleText}>loading..</Text> : normalView}
        {/* {normalView} */}
      </View>
    );
  }

}

const styles = StyleSheet.create({
  titleText: {
    fontSize: 20,
    fontWeight: "bold"
  },
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
  tinyLogo: {
    width: 50,
    height: 50,
  },
})
export default Profile

