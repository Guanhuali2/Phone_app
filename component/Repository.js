import React, { Component } from 'react';
import { StyleSheet, Text, View } from 'react-native';
import {RepositoryFetch} from './RepositoryFetch.js'

/*
Repository view class
*/
class Repository extends Component {
  constructor(props) {
    super(props);
    this.state = {
      loading: true
    }
    const accessToken = '';
    this.res = new RepositoryFetch(accessToken);
    this.getFile();
  }
  /*
  get fetch result from model class, change state
  */
  async getFile(){
    const temp = await this.res.file();
    //if error found , report to the class state
    if (temp == "failed"){
      this.setState({ hasError: true });
      return;
    }
    const body = this.res.getfile();
    this.setState({
      loading:false
    })
    //if error found report to the class state
    try{
    this.setState({
      responsitory: body.responsitory
    })
    }
    catch(error){
      this.setState({ hasError: true });
    }
  }
  /*
  iteration to create list
  */
  iteratedict() {
    if (!this.state.responsitory) return null;
    return Object.keys(this.state.responsitory).map((keys) => (
      <View key={keys +'0'}  style={styles.container}>        
        <Text key= {keys +'4'} style={styles.titleText}>{this.state.responsitory[keys].name}</Text>  
        <Text key= {keys +'3'} style={styles.titleText}>{this.state.responsitory[keys].url}</Text>  
        <Text key= {keys +'2'} style={styles.titleText}>{this.state.responsitory[keys].owner.login}</Text>  
        <Text key= {keys +'1'} style={styles.titleText}>{this.state.responsitory[keys].description}</Text>  
      </View>
    )
    )
  }

  render() {
    let normalView;
    //if error found, print invalid page
    if(this.state.hasError){
      normalView = (<View keys = "haha">
        <Text key='names' style={styles.titleText}>Invalid Profile Page</Text>
      </View>)
    }else{
      normalView = (          
      <View keys="haha2" style={styles.container}>
        {this.iteratedict()}
      </View>)
    }

    return (
      <View keys="laji" style={styles.container}>
        {this.state.loading ? <Text style={styles.titleText}>loading..</Text> : normalView}
      </View>
    );
  }

}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
  tinyLogo: {
    width: 50,
    height: 50,
  },
  titleText: {
    fontSize: 12,
    fontWeight: "bold"
  }
})
export default Repository