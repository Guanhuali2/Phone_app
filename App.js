import * as React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import Profile from './component/Profile.js'
import repositories from './component/Repository.js'
import { NavigationContainer } from '@react-navigation/native';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';

/*
Home page view
*/
function HomeScreen() {
  return(
    <View style={{flex: 1, justifyContent: 'center', alignItems: 'center'}}>
      <Text>Home</Text>
    </View>
  )
}

const Tab = createBottomTabNavigator();
/*
Nav bar
*/
export default function App() {
  return (
    <NavigationContainer>
      <Tab.Navigator>
        <Tab.Screen name="Home" component={HomeScreen} />
        <Tab.Screen name="Settings" component={Profile} />
        <Tab.Screen name="Res" component={repositories} />
      </Tab.Navigator>
    </NavigationContainer>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
