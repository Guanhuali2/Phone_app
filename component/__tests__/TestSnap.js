import 'react-native';
import React from 'react';
import Profile from '../Profile.js'
import Repository from '../Repository.js'
import App from '../../App.js'
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { render, waitFor } from '@testing-library/react-native';


jest.useFakeTimers()


test('renders correctly1', () => {
  const tree = renderer.create(
    <Profile />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly2', () => {
  const tree = renderer.create(
    <Repository />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly3', () => {
  const tree = renderer.create(
    <App />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

