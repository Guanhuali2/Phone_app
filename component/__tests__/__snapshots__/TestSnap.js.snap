// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly1 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  >
    loading..
  </Text>
</View>
`;

exports[`renders correctly2 1`] = `
<View
  keys="laji"
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 12,
        "fontWeight": "bold",
      }
    }
  >
    loading..
  </Text>
</View>
`;

exports[`renders correctly3 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <View
      pointerEvents="auto"
      style={
        Array [
          Object {
            "flex": 1,
            "overflow": "hidden",
          },
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
        ]
      }
    >
      <View
        collapsable={false}
        pointerEvents="auto"
        removeClippedSubviews={false}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text>
              Home!
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    pointerEvents="auto"
    style={
      Object {
        "backgroundColor": "rgb(255, 255, 255)",
        "borderTopColor": "rgb(216, 216, 216)",
        "borderTopWidth": 0.5,
        "bottom": 0,
        "elevation": 8,
        "height": 49,
        "left": 0,
        "paddingBottom": 0,
        "paddingHorizontal": 0,
        "position": null,
        "right": 0,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityLabel="Home, tab, 1 of 5"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "selected": true,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "textAlign": "center",
              },
              Object {
                "color": "rgb(0, 122, 255)",
              },
              Object {
                "fontSize": 10,
              },
              undefined,
            ]
          }
        >
          Home
        </Text>
      </View>
      <View
        accessibilityLabel="Profile, tab, 2 of 5"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "selected": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "textAlign": "center",
              },
              Object {
                "color": "#8E8E8F",
              },
              Object {
                "fontSize": 10,
              },
              undefined,
            ]
          }
        >
          Profile
        </Text>
      </View>
      <View
        accessibilityLabel="Repository, tab, 3 of 5"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "selected": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "textAlign": "center",
              },
              Object {
                "color": "#8E8E8F",
              },
              Object {
                "fontSize": 10,
              },
              undefined,
            ]
          }
        >
          Repository
        </Text>
      </View>
      <View
        accessibilityLabel="Followers, tab, 4 of 5"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "selected": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "textAlign": "center",
              },
              Object {
                "color": "#8E8E8F",
              },
              Object {
                "fontSize": 10,
              },
              undefined,
            ]
          }
        >
          Followers
        </Text>
      </View>
      <View
        accessibilityLabel="Followering, tab, 5 of 5"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "selected": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "textAlign": "center",
              },
              Object {
                "color": "#8E8E8F",
              },
              Object {
                "fontSize": 10,
              },
              undefined,
            ]
          }
        >
          Followering
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
